tests/Equinox.MessageDb.Integration/AdjacentProjection.fs

```diff
@@ -0,0 +1,130 @@
+module Equinox.MessageDb.Integration.AdjacentProjection
+
+open Dapper
+open Domain
+open Equinox.MessageDb
+open Swensen.Unquote
+open Xunit
+
+let defaultBatchSize = 500
+let connectionString = "Host=localhost; Username=postgres; Password=; Database=message_store; Port=5432; Maximum Pool Size=10; Search Path=message_store, public"
+let connectToLocalStore _ =
+  MessageDbClient(connectionString)
+type Context = MessageDbContext
+type Category<'event, 'state, 'req> = MessageDbCategory<'event, 'state, 'req>
+
+let createContext connection batchSize = Context(connection, batchSize = batchSize)
+
+
+module Projection =
+    [<CLIMutable>]
+    type ContactPreferences = { id: string; many_promotions: bool; little_promotions: bool; product_review: bool; quick_surveys: bool }
+    let createTable (conn: Npgsql.NpgsqlConnection) =
+        conn.Execute("create table if not exists public.contact_preferences (
+            id text not null primary key,
+            many_promotions bool not null default false,
+            little_promotions bool not null default false,
+            product_review bool not null default false,
+            quick_surveys bool not null default false
+        )") |> ignore
+
+    let project streamId (state: ContactPreferences.Fold.State) (conn: Npgsql.NpgsqlConnection) = task {
+        let! _ = conn.ExecuteAsync(
+            "insert into public.contact_preferences (id, many_promotions, little_promotions, product_review, quick_surveys)
+             values (@id, @many_promotions, @little_promotions, @product_review, @quick_surveys)
+             on conflict (id) do update
+             set many_promotions = EXCLUDED.many_promotions,
+                 little_promotions = EXCLUDED.little_promotions,
+                 product_review = EXCLUDED.product_review,
+                 quick_surveys = EXCLUDED.quick_surveys",
+            { id = FsCodec.StreamId.toString streamId; many_promotions = state.manyPromotions; little_promotions = state.littlePromotions; product_review = state.productReview; quick_surveys = state.quickSurveys })
+        return () }
+    let readOne id (conn: Npgsql.NpgsqlConnection) = task {
+        let! result = conn.QueryFirstOrDefaultAsync<ContactPreferences>("select * from public.contact_preferences where id = @id", {| id = id |})
+        return if System.Object.ReferenceEquals(result, null)
```

Comment:
if obj.ReferenceEquals()
but I think it can be `Option.ofObj`

Reply:
> Option.ofObj
nope, F# thinks the type can't be null, but the CLR begs to differ

Reply:
I wonder does AllowNullLiteral fix that? or box?